lib/tanker/identity.rb

```diff
@@ -117,14 +114,14 @@ def self.get_public_identity(serialized_identity)
       public_identity = {}
       public_keys.each { |key| public_identity[key] = identity.fetch(key) }
 
-      if identity['target'] == 'email'
-        identity['target'] = 'hashed_email'
-        public_identity['value'] = Base64.strict_encode64(Crypto.generichash(public_identity['value'], BLOCK_HASH_SIZE))
+      public_identity['target'] = 'hashed_email' if identity['target'] == 'email'
+      if identity['target'] != 'user'
+        public_identity['value'] = Crypto.hashed_provisional_value(public_identity['value'])
```

Comment:
Mixing conditionals in "standard" and "modifier" (`code if condition`) forms feels a bit unfortunate for conditions at the same level.

Maybe pick between one of:

```ruby
# It's a bit long
public_identity['target'] = 'hashed_email' if identity['target'] == 'email'
public_identity['value'] = Crypto.hashed_provisional_value(public_identity['value']) unless identity['target'] == 'user'
```

```ruby
# It's ok
if identity['target'] == 'email'
  public_identity['target'] = 'hashed_email'
end

if identity['target'] != 'user'
  public_identity['value'] = Crypto.hashed_provisional_value(public_identity['value'])
end
```

```ruby
# It's ok too!
if identity['target'] != 'user'
  public_identity['target'] = 'hashed_email' if identity['target'] == 'email'
  public_identity['value'] = Crypto.hashed_provisional_value(public_identity['value'])
end
```

Reply:
I'll take the 2nd one, much to my IDE's displeasure, the lines get a bit long otherwise!

Reply:
where is the salt:
```ruby

salt = Crypto.generichash(identity['private_signature_key'], Crypto::BLOCK_HASH_SIZE)
public_identity['value'] = Crypto.hashed_provisional_value(salt + public_identity['value'])
```

Reply:
Added the salt